src/script/main/app.ts

```diff
@@ -433,14 +443,24 @@ export class App {
 
       if (supportsMLS()) {
         // Once all the messages have been processed and the message sending queue freed we can now:
-
         const mlsConversations = conversations.filter(isMLSConversation);
 
+        //initialize all the mls groups that are already established (schedule periodic key updates, etc.)
+        await initialiseEstablishedMLSCapableConversations(this.core);
+
         //add the potential `self` and `team` conversations
         await registerUninitializedSelfAndTeamConversations(mlsConversations, selfUser, clientEntity().id, this.core);
 
+        //join all the mls groups that are known by the user but were migrated to mls
+        await joinConversationsAfterMigrationFinalisation({
+          updatedConversations: conversations,
+          initialDatabaseConversations: initialLocalConversations,
+          core: this.core,
+          conversationRepository: conversationRepository,
+        });
+
         //join all the mls groups we're member of and have not yet joined (eg. we were not send welcome message)
-        await initMLSConversations(mlsConversations, this.core);
+        await joinNewMLSConversations(mlsConversations, this.core);
```

Comment:
- first we initialise already existing mls conversations (so schedule periodic key materials)
- then we register team and self conversations if they havent been yet established
- then we join the mls conversations that have been already migrated to mls (user was not supporting mls when protocol change events or welcome messages were sent)
- in the end we join all new mls conversations we are members of but have not yet joined on the current client